test/unit-test/test_derivation.ml

```diff
@@ -135,24 +135,19 @@ let%expect_test "derivation" =
          Empty))))
   |}];
 
-  print Context.empty (Abs (["x"], Var "x"));
+  print Context.empty (Abs (["X"], Var "X"));
   [%expect {|
-    (Ok (
-      (TyVar b) (Eq (TyVar b) (TyConstraint (TyFun ((TyVar a)) (TyVar a)) Empty)))) |}];
+    (Ok ((TyFun ((TyVar a)) (TyVar a)) Empty)) |}];
```

Comment:
I checked this derivation with paper.
![](https://pbs.twimg.com/media/DuDBCXVVAAATZKa.jpg)